### PR TITLE
Add ExtendedFields model

### PR DESCRIPTION
**Pull request by TaiWilkin**
## Overview

Adds the ExtendedFields model, the first step towards extended profiles.

The `field_name`, describing the field name for the extended field,
is pulled from a list of choices. Currently, available field_names are
country, address, name, number_of_workers and native_language_name.

The `value` is stored in a jsonb object, with different structure for
different field types. Numeric fields will be stored as {“min”: v1,
“max”: v2}. If there is a single numeric value, we will set both min
and max to it. If there is no provided value, we won't make an
ExtendedField row.

Each ExtendedField will be created from either a facility claim or a
facility list item. It will be must linked to one or the other, but
not both.

Connects #1492 

## Demo

Screen Shot 2021-10-26 at 2 16 20 PM

## Testing Instructions

* Run `./scripts/manage migrate` and confirm the migration is successful.
* Run `./scripts/manage shell_plus` and create an ExtendedField (replace the facility and list_item ids with ones from your database).
```
value = { "min": 100, "max": 500 }
ExtendedField.objects.create(contributor_id=3, facility_id='CN2021299G038W1', facility_list_item_id='372', field_name='number_of_workers', value=value)
```
* Confirm that the Extended field was created successfully and has the correct data content.  
```
ExtendedField.objects.get(id=1).__dict__
```
## Checklist

- [x] `fixup!` commits have been squashed
- [x] CI passes after rebase
- [x] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines

### CODE REVIEW

File: src/django/api/models.py [hunk 1]
```diff
@@ -2152,6 +2152,72 @@ def __str__(self):
...
 
+class ExtendedField(models.Model):
+    """
+    Extended data fields available to include on facilities.
+    Fields will be related to either a claim or list item; they must reference
+    one, but not both.
+    """
+    COUNTRY = 'country'
+    NAME = 'name'
+    ADDRESS = 'address'
```
[the remark this answers is not in the document]
Comment (TaiWilkin) [reply src/django/api/models.py:2163]: Yes, that's my understanding as well.

File: src/django/api/models.py [hunk 2]
```diff
@@ -2152,6 +2152,72 @@ def __str__(self):
...
+        default=False,
+        null=False,
+        help_text='Whether or not this field has been verified.'
+    )
+    field_name = models.CharField(
+        max_length=200,
+        null=False,
+        blank=False,
+        choices=FIELD_CHOICES,
+        default=COUNTRY,
```
[the remark this answers is not in the document]
Comment (TaiWilkin) [reply src/django/api/models.py]: Good call.

**Comment by TaiWilkin** [comment]
@jwalgran Ready for another look.

**Comment by jwalgran** [comment]
Looking at this now.

**Review by jwalgran — APPROVED** [review]

**Comment by TaiWilkin** [comment]
Thank you for reviewing!

**Pull request merged**